src/libraries/System.Private.CoreLib/src/System/Threading/NativeRuntimeEventSource.PortableThreadPool.NativeSinks.cs

```diff
@@ -182,6 +195,18 @@ public void ThreadPoolIOEnqueue(RegisteredWaitHandle registeredWaitHandle)
             LogThreadPoolIODequeue(NativeOverlapped, Overlapped, ClrInstanceID);
         }
 
+        [NonEvent]
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public unsafe void ThreadPoolIODequeue(NativeOverlapped* nativeOverlapped)
+        {
+            if (IsEnabled(EventLevel.Verbose, Keywords.ThreadingKeyword | Keywords.ThreadTransferKeyword))
+            {
+                ThreadPoolIODequeue(
+                    (IntPtr)nativeOverlapped,
+                    (IntPtr)OverlappedData.GetOverlappedFromNative(nativeOverlapped).GetHashCode());
```

Comment:
Is this more expensive than what happened before?

Reply:
Not much, previously the pointer to the managed object was reported, we have switched away from that in other similar cases and used the hash code instead because managed objects can be moved around.